utils/src/types.rs

```diff
@@ -0,0 +1,40 @@
+use crate::diff_score;
+use ethers_core::abi::Abi;
+use ethers_solc::ArtifactId;
+use std::collections::BTreeMap;
+
+type ArtifactWithContractRef<'a> = (&'a ArtifactId, &'a (Abi, Vec<u8>));
+
+/// Wrapper type that maps an artifact to a contract ABI and bytecode.
+pub type ContractsByArtifact = BTreeMap<ArtifactId, (Abi, Vec<u8>)>;
+
+pub trait ContractsByArtifactExt {
```

Comment:
not a fan of type alias + extension trait combination

let's just make this a new type that's also deref

like 

```rust
struct ContractsByArtifact(BTreeMap<ArtifactId, (Abi, Vec<u8>)>);
```

or

```rust
struct ContractsByArtifact{
  inner: BTree....
}
```

and then impl Deref for ContractsByArtifact

Reply:
@joshieDo agree with Matt here